gns3server/compute/docker/docker_vm.py

```diff
@@ -342,6 +342,11 @@ def _create_network_config(self):
         # Give the information to the container the list of volume path mounted
         params["Env"].append("GNS3_VOLUMES={}".format(":".join(self._volumes)))
 
+        # Pass user configured for image to init script
+        if image_infos.get("Config", {"User": ""}).get("User"):
+            params["User"] = "root"
```

Comment:
Just a question. Why is root needed here?

Reply:
This tells docker to start the user inside docker container as root, so when the "init.sh" script runs it can configure the network. This is normally the default.

However, some images that are hardened for production use explicitly change the user inside the docker image to a non-root user which prevents the init.sh script from configuring the interfaces and hence does not work in GNS3.